hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -190,6 +191,13 @@ public KeyValueHandler(ConfigurationSource config,
     byteBufferToByteString =
         ByteStringConversion
             .createByteBufferConversion(isUnsafeByteBufferConversionEnabled);
+
+    if (ContainerLayoutVersion.getConfiguredVersion(conf) ==
+        ContainerLayoutVersion.FILE_PER_CHUNK) {
+      LOG.warn("FILE_PER_CHUNK layout is deprecated. Update to FILE_PER_BLOCK.");
+      OzoneConfiguration.of(conf).set(ScmConfigKeys.OZONE_SCM_CONTAINER_LAYOUT_KEY,
+          "FILE_PER_BLOCK");
```

Comment:
If we are setting `OZONE_SCM_CONTAINER_LAYOUT_KEY` to `FILE_PER_BLOCK` in the code, we are actually removing support for `FILE_PER_CHUNK` not deprecating it.

With this change, there is no way to use `FILE_PER_CHUNK` anymore. Can we at least log a clear WARN message stating that the value of `OZONE_SCM_CONTAINER_LAYOUT_KEY` is getting overwritten in the code?

Reply:
This PR meant to disallow creating new FILE_PER_CHUNK containers, but it would still permit reading existing FILE_PER_CHUNK containers.

I'd love to get rid of FILE_PER_CHUNK code altogether but I'm not sure if anyone still have FILE_PER_CHUNK containers.